### PR TITLE
611 - Teaming up.md- Agility content

### LINKED ISSUE #611: Content about agile practices during team swarming phases of engagements in CSE
A section under collaboration folder might be a good place for this content

### PR DESCRIPTION
**Pull request by gldnblgty**
## What are you trying to address

- Content about some agility practices in team development

## Description of new changes

- docs/agile-development/teaming-up.md

Closes #611

### CODE REVIEW

**Review by stanchat — APPROVED** [review]
Looks good.  I will approve with some comments

I would replace **"Grooming"** with _**Refinement**_.  Grooming actually has a negative connotation
https://www.scrum.org/forum/scrum-forum/5635/grooming-illegal

Also I would spell out "US" to User Stories just to make it more clearer

*******************************************************************************
Backlog **Refinement** meetings
  - Size of stories (Too big? Too small?)
  - Are **User Stories** and tasks clear ?
  - Is Acceptance **Criteria(s)** enough and right?
  - Is everyone ready-to-go after taking the User Story/Task?

File: docs/agile-development/collaboration/teaming-up.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Identification phase accelerates the process of building a safe environment for every individual in the team, later on team has the required assets to follow the plan.
+And it is team's itself responsibility (SEs, TPM, Lead) to debug their Agility level.
+
+In every team stabilization takes time and pro-active agile debugging is the best accelerator to decrease the distraction away from sprint/engagement goal.
+
+Just as an example, agility debugging activities may include:
+
+- Dashboards related with "Goal" such as burndown/burnout, Item/PR Aging, Mood Chart ..etc. are accessible to the team and team is always up-to-date
+
+- Backlog Grooming meetings
```
Comment (madossan01) [inline docs/agile-development/collaboration/teaming-up.md:45]: For Backlog Refinement, does it make sense to point out to the Backlog Refinement section from code-with playbook?  Please consider also to move the questions to the backlog refinement document. This way the Backlog Refinement section becomes the source of truth for the refinement practice.

File: docs/agile-development/collaboration/teaming-up.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  - Size of stories (Too big? Too small?)
+  - Are US and tasks clear ?
+  - Are Acceptance Criterias enough and right?
+  - Is everyone ready-to-go after taking the User Story/Task?
+
+- Running [Efficient Retrospectives](../retrospectives/readme.md)
```
Comment (madossan01) [inline docs/agile-development/collaboration/teaming-up.md:51]: Please consider moving the questions to the retrospectives document so the Retrospective section becomes the source of truth for the practice.

File: docs/agile-development/collaboration/teaming-up.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+- Is the Sprint Goal clear in every iteration ?
+
+- Is the estimation process in the team improving over time or does it meet the delivery/workload prediction?
+
+Kindly check [Scrum Values](https://scrumguides.org/scrum-guide.html) to have a better understanding to improve team commitment.
```
Comment (madossan01) [inline docs/agile-development/collaboration/teaming-up.md:57]: Very good reminder.

File: docs/agile-development/collaboration/teaming-up.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+- Identification of "Software Craftspersonship" areas which means the tools and methods will be widely used during the engagement and taking the required actions on team upskilling side if necessary.
+
+  - Github, VSCode LiveShare, AzDevOps, necessary development tools & libraries ... more.
+
+- Identification of communication channels, feedback loops and recurrent team call slots out of regular sprint meetings
+
+- Introduction to [Technical Agility Team Manifesto](../team-agreements/team-manifesto/readme.md) and planning the technical delivery by aiming to keep
+technical debt risk minimum.
+
+## Following the Plan and Agile Debugging
```
Comment (madossan01) [inline docs/agile-development/collaboration/teaming-up.md:34]: Team needs and configuration may change throughout development. In this case, do you think it is worth add a reminder in the section to review and keep the plan up to date according to the team changes/needs?

**Review by madossan01 — CHANGES_REQUESTED** [review]

**Comment by gldnblgty** [comment]
> Looks good. I will approve with some comments
> 
> I would replace **"Grooming"** with _**Refinement**_. Grooming actually has a negative connotation
> https://www.scrum.org/forum/scrum-forum/5635/grooming-illegal
> 
> Also I would spell out "US" to User Stories just to make it more clearer
> 
> Backlog **Refinement** meetings
> 
> * Size of stories (Too big? Too small?)
> * Are **User Stories** and tasks clear ?
> * Is Acceptance **Criteria(s)** enough and right?
> * Is everyone ready-to-go after taking the User Story/Task?

thank you for your time on review @stanchat ! appreciated for the comments and i fixed them in 67ebc9d83f919bfdbbc87e9581bcfe37928c57d5 and aa81395b2f71efd8f75363737fd6072c19659458

File: docs/agile-development/collaboration/teaming-up.md (hunk 2 above)
Comment (gldnblgty) [reply docs/agile-development/collaboration/teaming-up.md:51]: thank you very much for this suggestion :) 
I will do follow-ups on it.

File: docs/agile-development/collaboration/teaming-up.md (hunk 4 above)
Comment (gldnblgty) [reply docs/agile-development/collaboration/teaming-up.md:35]: added below part in 5b86b7bd3759f4ede15c1e9d53c13b0a12cf4867
> Team is also responsible to keep the plan up-to-date based on team changes/needs and debugging results.

File: docs/agile-development/collaboration/teaming-up.md (hunk 1 above)
Comment (gldnblgty) [reply docs/agile-development/collaboration/teaming-up.md]: added in 5b86b7bd3759f4ede15c1e9d53c13b0a12cf4867
thanks @madossan01! :)

File: docs/agile-development/collaboration/teaming-up.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+Identification phase accelerates the process of building a safe environment for every individual in the team, later on team has the required assets to follow the plan.
+And it is team's itself responsibility (SEs, TPM, Lead) to debug their Agility level.
```
[the remark this answers is not in the document]
Comment (gldnblgty) [reply docs/agile-development/collaboration/teaming-up.md]: fixed in 5b86b7bd3759f4ede15c1e9d53c13b0a12cf4867

File: docs/agile-development/collaboration/teaming-up.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+- Identification of styles/preferences in communication, sharing, learning, decision making of each team member
+  
+  - Talking about necessity of pair programming
+  - Decisions on backlog management & grooming meetings, weekly design sessions, social time sessions...etc.
+  - Sync/Async communication methods, work hours/flexible times
+
+- Decisions and identifications of charts that will be helpful to provide transparent and true information to everyone
+
+- Identification of "Software Craftspersonship" areas which means the tools and methods will be widely used during the engagement and taking the required actions on team upskilling side if necessary.
```
[the remark this answers is not in the document]
Comment (gldnblgty) [reply docs/agile-development/collaboration/teaming-up.md:25]: awesome suggestion. added in 5b86b7bd3759f4ede15c1e9d53c13b0a12cf4867

File: docs/agile-development/collaboration/teaming-up.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+In order to minimize the risk and set the exceptations on the right way for all parties, an identification phase is important to understand each other.
+Some potential steps in this phase may be as following (not limited):
+
+- [Working agreement](../team-agreements/working-agreements/readme.md)
+
+- Identification of styles/preferences in communication, sharing, learning, decision making of each team member
+  
+  - Talking about necessity of pair programming
+  - Decisions on backlog management & grooming meetings, weekly design sessions, social time sessions...etc.
```
[the remark this answers is not in the document]
Comment (gldnblgty) [reply docs/agile-development/collaboration/teaming-up.md]: true, @stanchat raised the same one also.
I fixed wording in 67ebc9d. thanks for filling the gap on my side about this CSE guidance ! :)

File: docs/agile-development/collaboration/teaming-up.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+## Identification
+
+As it's stated in Tuckman's [team phases](https://en.wikipedia.org/wiki/Tuckman%27s_stages_of_group_development), traditional team development has several stages.
+However those phases can be extremely fast or sometimes mismatched in CSE engagement teams due to external factors.
+
+In order to minimize the risk and set the exceptations on the right way for all parties, an identification phase is important to understand each other.
+Some potential steps in this phase may be as following (not limited):
+
+- [Working agreement](../team-agreements/working-agreements/readme.md)
```
[the remark this answers is not in the document]
Comment (gldnblgty) [reply docs/agile-development/collaboration/teaming-up.md]: fixed 5b86b7bd3759f4ede15c1e9d53c13b0a12cf4867

File: docs/agile-development/collaboration/teaming-up.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
+# Engagement Team Development in CSE
+
+In every CSE engagement, dynamics are different so are the team requirements. Based on transfer learning among teams, we aim to build right "code-with" environments in every team.
+
+This documentation gives a high-level template with some suggestions by aiming to accelerate team swarming phase to achieve a high speed agility however it has no intention to provide a list of "must-do" items.
+
+## Identification
+
+As it's stated in Tuckman's [team phases](https://en.wikipedia.org/wiki/Tuckman%27s_stages_of_group_development), traditional team development has several stages.
+However those phases can be extremely fast or sometimes mismatched in CSE engagement teams due to external factors.
```
[the remark this answers is not in the document]
Comment (gldnblgty) [reply docs/agile-development/collaboration/teaming-up.md]: i added this in 5b86b7bd3759f4ede15c1e9d53c13b0a12cf4867 , thank you!!

File: docs/agile-development/collaboration/teaming-up.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
+# Engagement Team Development in CSE
```
[the remark this answers is not in the document]
Comment (gldnblgty) [reply docs/agile-development/collaboration/teaming-up.md]: very good one, fixed in 5b86b7bd3759f4ede15c1e9d53c13b0a12cf4867

**Review by madossan01 — APPROVED** [review]

**Pull request merged**